DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/README.md

```diff
@@ -1,257 +1,286 @@
 
-Bilateral filters are non-linear and therefore non-separable. In the case of a 5x5 window (shown below), only 9 pixels (not 25) are used in the computation; this is an artifact of the Bayer image format. The most accurate approach would produce the bilateral filter window and convolve the entire window of the given pixel color (for the image below, red) at once to generate the output pixel (the middle pixel). For the 5x5 case, this would result in 9 multiplications that need to be summed. <br/>
+Use the Linux instructions to build and run the program.
 
-<img src="conv.png" alt="conv" width="400"/>
+You can specify a FPGA runtime node using a single line script similar to the following example.
 
-This produces a long chain of adders to sum the results of the multiplications. In this design, we approximate the bilateral filter by making it separable. We first apply a 1D vertical filter to the middle pixel, and then a 1D horizontal filter, as shown in the image below. This reduces the number of multiplications that need to be summed together to 3. In our design, we apply the vertical filter to *all* pixels first, and then apply the horizontal filter, which results in the *corner* pixels indirectly applying some weight to the middle pixel.
+```
+qsub -I -l nodes=1:fpga_runtime:ppn=2 -d .
+```
 
-<img src="bilateral_estimate.png" alt="bilateral_estimate" width="900"/>
+- `-I` (upper case I) requests an interactive session.
+- `-l nodes=1:fpga_runtime:ppn=2` (lower case L) assigns one full node.
+- `-d .` makes the current folder as the working directory for the task.
 
-### ANR FPGA Design
-The ANR algorithm is designed as a streaming kernel system with input pixels streaming through the input pipe, and the denoised output pixels streaming out the output pipe, as shown in the figure below. The design consists of two kernels, `Vertical Kernel` and `Horizontal Kernel`, that are connected by an internal SYCL pipe, as shown in the figure below. The `Vertical Kernel` computes an intensity sigma value based on the current pixel, computes the bilateral filter, and applies it to the current window to produce an intermediate pixel value. The `Vertical Kernel` kernel sends three values through the internal pipe: the original pixel value, the current pixel value (i.e., intermediate pixel that was just computed), and the intensity sigma value. The `Horizontal Kernel` streams in these tuples and performs a similar computation but on a horizontal window. It uses the forwarded intensity sigma value to compute the bilateral filer, the new pixel values to perform the bilateral filter computation, and the original pixel to perform *alpha blending*, where the output pixel is a weighted percentage of the original pixel value and the denoised pixel value.
+  |Available Nodes    |Command Options
+  |:---               |:---
+  |FPGA Compile Time  |`qsub -l nodes=1:fpga_compile:ppn=2 -d .`
+  |FPGA Runtime       |`qsub -l nodes=1:fpga_runtime:ppn=2 -d .`
```

Comment:
One suggestion given by one of the FAEs I talked to regarding these readme changes would be to have the FPGA runtime line split in two with the commands to get an Arria 10 FPGA and a Stratix 10 FPGA:
```suggestion
  |FPGA Runtime (Arria 10)       |`qsub -l nodes=1:fpga_runtime:arria10:ppn=2 -d .`
  |FPGA Runtime (Stratix 10)       |`qsub -l nodes=1:fpga_runtime:stratix10:ppn=2 -d .`
```